clustered/common/src/main/java/org/ehcache/clustered/common/PoolAllocation.java

```diff
@@ -72,6 +74,24 @@ public long getSize() {
     public String getResourceName() {
       return resourceName;
     }
+
+    @Override
+    public boolean isCompatible(PoolAllocation other) {
```

Comment:
IMHO compatibility check makes sense on `ServerStoreConfiguration`, rather than anywhere else. As I understand its the two configs which we want to see if they are compatible and based on that the pool allocations are compatible or not. I would rather have compatibility check somewhere else.